Source/Engine/McBopomofoLM.cpp

```diff
@@ -0,0 +1,92 @@
+#include "McBopomofoLM.h"
+#include <algorithm>
+#include <iterator>
+#include <unordered_set>
+
+using namespace McBopomofo;
+
+McBopomofoLM::McBopomofoLM()
+{
+}
+
+McBopomofoLM::~McBopomofoLM()
+{
+    m_languageModel.close();
+    m_userPhrases.close();
+    m_excluddePhrases.close();
+}
+
+void McBopomofoLM::loadLanguageModel(const char* languageModelDataPath)
+{
+    m_languageModel.close();
+    m_languageModel.open(languageModelDataPath);
+}
+
+void McBopomofoLM::loadUserPhrases(const char* userPhrasesDataPath,
+                                   const char* excludedPhrasesDataPath)
+{
+    m_userPhrases.close();
+    m_userPhrases.open(userPhrasesDataPath);
```

Comment:
`LanguageModelManager` passes `""` to `userPhrasesDataPath` when creating Plain Bopomofo's LM. Perhaps allow `nullptr` here, and only call `open` if `userPhrasesDataPath` is not `nullptr`?

Same for `excludedPhrasesDataPath`.

I realize calling `::open("", O_RDONLY)` in `FastLM` will likely lead to a safely-guarded failure, but one more guard shouldn't hurt…